Proposals/NNNN-calendar-improvements.md

```diff
@@ -0,0 +1,261 @@
+# Calendar Sequence Enumeration
+
+* Proposal: [SF-NNNN](NNNN-calendar-improvements.md)
+* Authors: [Tony Parker](https://github.com/parkera)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Implementation: [Pull Request](https://github.com/apple/swift-foundation/pull/322)
+* Review: [Pitch](https://forums.swift.org/t/pitch-calendar-sequence-enumeration/68521)
+
+## Introduction
+
+In macOS 14 / iOS 17, `Calendar` was rewritten entirely in Swift. One of the many benefits of this change is that we can now more easily create Swift-specific `Calendar` API that feels more natural than the existing `enumerate` methods. In addition, we are taking the opportunity to add a new field to the `DateComponents` type to handle one case that was only exposed via the somewhat esoteric CoreFoundation API `CFCalendarDecomposeAbsoluteTime`.
+
+## Motivation
+
+The existing `enumerateDates` method on `Calendar` is basically imported from an Objective-C implementation. We can provide much better integration with other Swift API by providing a `Sequence`-backed enumeration. The `nextDate` API can similarly be improved with a `Sequence` API.
+
+We also need to support a new `dayOfYear` field on `DateComponents` to support a Swift-only implementation of `ISO8601FormatStyle` in `FoundationEssentials`.
+
+## Proposed solution
+
+We propose a new field on `DateComponents` and associated options / units:
+
+```swift
+extension Calendar {
+    public enum Component : Sendable {
+        // .. existing fields
```

Comment:
Is it useful commentary  ?
```suggestion
```

Reply:
I do this so it's clear we are not adding this `Component` type as part of this proposal.